templates/server.conf.erb

```diff
@@ -338,3 +379,60 @@ host: <%= scope['pulp::email_host'] %>
 port: <%= scope['pulp::email_port'] %>
 from: <%= scope['pulp::email_from'] %>
 enabled: <%= scope['pulp::email_enabled'] %>
+
+# = Lazy =
+#
+# Settings for lazy content loading.
+#
+# enabled:
+#   This controls whether redirect is enabled
+#
+# redirect_host:
+#   The host FQDN or IP to which requests are redirected.
+#
+# redirect_port:
+#   The TCP port to which requests are redirected
+#
+# redirect_path:
+#   The base path to which requests are redirected
+#
+# https_retrieval:
+#   boolean; controls whether Pulp uses HTTPS or HTTP to
+#   retrieve content from the streamer.
+#   WARNING: Setting this to 'false' is not safe if you wish
+#            to use Pulp to provide repository entitlement
+#            enforcement. It is strongly recommended to keep
+#            this set to 'true' and use certificates that are
+#            signed by a trusted authority on the web server
+#            that serves as the streamer reverse proxy.
+#
+# download_interval:
+#   The interval in minutes between checks for content cached
+#   by the Squid proxy.
+#
+# download_concurrency:
+#   The number of downloads to perform concurrently when
+#   downloading content from the Squid cache.
+
+[lazy]
+<% if scope['pulp::lazy_enabled'] != :undef -%>
+enabled: <%= scope['pulp::lazy_enabled'] %>
+<% end -%>
+<% if scope['pulp::lazy_redirect_host'] != :undef -%>
+redirect_host: <%= scope['pulp::lazy_redirect_host'] %>
+<% end -%>
+<% if scope['pulp::lazy_redirect_port'] != :undef -%>
+redirect_port: <%= scope['pulp::lazy_redirect_port'] %>
+<% end -%>
+<% if scope['pulp::lazy_redirect_path'] != :undef -%>
+redirect_path: <%= scope['pulp::lazy_redirect_path'] %>
+<% end -%>
+<% if scope['pulp::lazy_https_retrieval'] != :undef -%>
+https_retrieval: <%= scope['pulp::lazy_https_retrieval'] %>
+<% end -%>
+<% if scope['pulp::lazy_download_interval'] != :undef -%>
+download_interval: <%= scope['pulp::lazy_download_interval'] %>
+<% end -%>
+<% if scope['pulp::lazy_download_concurrency'] != :undef -%>
+download_concurrency: <%= scope['pulp::lazy_download_concurrency'] %>
+<% end -%>
```

Comment:
i think it would make more sense to leave out the others if lazy_enabled is false and include them all (with sane defaults) if its is enabled.